From d16497e49cad964ebf49961eded5fe4e90fdc9ba Mon Sep 17 00:00:00 2001
From: Andrew Sutherland <dr3wsuth3rland@gmail.com>
Date: Mon, 17 Dec 2012 14:43:24 -0600
Subject: [PATCH 2/2] RIL: fix signal for legacy gsm devices

our parcels are short causing mLteRssnr to be 0 which  is a valid
value and causes SignalStrength to think we are an LTE device.

implement as oldRil "signalstrengthgsm"

Change-Id: Id1b47dbee6cc875bc2cde2f499f6ba788fa9ccb5
---
 src/java/com/android/internal/telephony/RIL.java | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/src/java/com/android/internal/telephony/RIL.java b/src/java/com/android/internal/telephony/RIL.java
index a5f4ea7..8e8bdbc 100644
--- a/src/java/com/android/internal/telephony/RIL.java
+++ b/src/java/com/android/internal/telephony/RIL.java
@@ -4025,9 +4025,17 @@ public class RIL extends BaseCommands implements CommandsInterface {
 
     protected Object
     responseSignalStrength(Parcel p) {
-        // Assume this is gsm, but doesn't matter as ServiceStateTracker
-        // sets the proper value.
-        SignalStrength signalStrength = SignalStrength.makeSignalStrengthFromRilParcel(p);
+        SignalStrength signalStrength;
+        if (needsOldRilFeature("signalstrengthgsm")) {
+            int gsmSignal  = p.readInt();
+            int gsmErrRate = p.readInt();
+            signalStrength = new SignalStrength(gsmSignal, gsmErrRate,
+                                                -1, -1, -1, -1, -1, true);
+        } else {
+            // Assume this is gsm, but doesn't matter as ServiceStateTracker
+            // sets the proper value.
+            signalStrength = SignalStrength.makeSignalStrengthFromRilParcel(p);
+        }
         return signalStrength;
     }
 
-- 
1.8.1.2

